File: ProgramacionMultimedia/repaso1/app/src/main/java/com/example/repaso1/Alumno.java
package com.example.repaso1;

import java.util.ArrayList;

public class Alumno {
    private int nExpediente;//representa el número de cuenta.
    private String Nombre;
    private String Apellido;
    private ArrayList<Falta> Faltas=new ArrayList<Falta>();

    public Alumno(int ne,String nom, String ape)
    {this.Nombre=nom;
        this.Apellido=ape;
        this.nExpediente=ne;
    }

    public int getnExpediente() {
        return nExpediente;
    }

    public String getNombre() {
        return Nombre;
    }

    public String getApellido() {
        return Apellido;
    }

    public void registrarFalta(int dia, int mes)
    {

        for(Falta f:Faltas)
            if(f.getDia()==dia&&f.getMes()==mes)
            {
                return;
            }
            Faltas.add(new Falta(dia, mes));
    }

    public ArrayList<ContadorFalta> getFaltas()
    {
        ArrayList<ContadorFalta> resultado = new ArrayList<ContadorFalta>();
        int []contaux=new int[12];

        for(Falta f:Faltas)
        {
            //Suma uno en el número del mes
            contaux[f.getMes()-1]++;
        }

        for(int i=0;i<contaux.length;i++)
        {
            resultado.add(new ContadorFalta(i+1, contaux[i]));
        }
        return resultado;

    }

}

File: Desarrollo de Interfaces/SOLUCIONES/WPF/WPFLista/WPFLista/Alumno.vb
Public Class Alumno
    Dim _Nombre As String
    Dim _Apellido As String
    Dim _Id As String

    Public Sub New(nombre As String, apellido As String, id As String)
        Me._Nombre = nombre
        Me._Apellido = apellido
        Me._Id = id
    End Sub

    Public Property Nombre As String
        Get
            Return _Nombre
        End Get
        Set(value As String)
            _Nombre = value
        End Set
    End Property

    Public Property Apellido As String
        Get
            Return _Apellido
        End Get
        Set(value As String)
            _Apellido = value
        End Set
    End Property

    Public Property Id As String
        Get
            Return _Id
        End Get
        Set(value As String)
            _Id = value
        End Set
    End Property
End Class
